tests/feature/test_outline.py

```diff
@@ -320,3 +320,58 @@ def _(site):
     result = pytester.runpytest("-s")
     result.assert_outcomes(passed=1)
     assert collect_dumped_objects(result) == ["https://my-site.com"]
+
+
+def test_variable_reuse(pytester):
+    """
+    Test example parameter name and step arg do not redefine each other's value
+    if the same name is used for both in different steps.
+    """
+
+    pytester.makefile(
+        ".feature",
+        outline=textwrap.dedent(
+            """\
+            Feature: Example parameters reuse
+                Scenario Outline: Check for example parameter re-use
+                    Given I get <param>
+                    When I set param to "other"
+                    Then I check <param>
+
+                    Examples:
+                        | param |
+                        | value |
+
+            """
+        ),
+    )
+
+    pytester.makepyfile(
+        textwrap.dedent(
+            """\
+            from pytest_bdd import given, when, then, parsers, scenarios
+            from pytest_bdd.utils import dump_obj
+
+            scenarios('outline.feature')
+
+
+            @given(parsers.parse('I get {param}'))
+            def _(param):
+                dump_obj(("param1", param))
+
+
+            @when(parsers.re('I set param to "(?P<param>.+)"'))
+            def _(param):
+                dump_obj(("param2", param))
```

Comment:
maybe better use steps like:
* `'param is initially set from the example table to "{value}"'`
* `'param is then by the step to to "{value}"'`

I'd find it easier to grasp